### PR TITLE
Add QueueSong Lambda and SQS Queue

### PR DESCRIPTION
**Pull request by agsaghhk**
Added QueueSong Lambda with SQS Queue event source to CDK stack. Related to #83 

### Testing
`npm run build`

`cdk synth` and confirmed that Lambda, SQS Queue, EventSource and IAM Role to allow SQS to trigger lambda are in output

### CODE REVIEW

File: cdk/lib/cdk-stack.ts [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
   constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
-    super(scope, id, props);
-
-    // The code that defines your stack goes here
+    super(scope, id, props)
 
+    // Can add timeout, deadLetterQueue, memorySize etc. later to fine tune the created lambda
+    const queueSongLambda = new lambda.Function( this, 'QueueSongLambda', {
+      code: new lambda.AssetCode(`../queue-song-lambda/src`),
+      handler: 'Foo.foo',
```
Comment (SaxyPandaBear) [inline cdk/lib/cdk-stack.ts:13]: This'll change when I actually finish rewriting the lambda, but fine as is atm

File: cdk/lib/cdk-stack.ts [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+      runtime: lambda.Runtime.JAVA_8,
+    })
+    const songRequestsQueue = new sqs.Queue(this, 'SongRequestQueue', {
+      visibilityTimeout: cdk.Duration.seconds(30), //default
+      receiveMessageWaitTime: cdk.Duration.seconds(20)
```
Comment (SaxyPandaBear) [inline cdk/lib/cdk-stack.ts:18]: This is a message delay before the lambda receives it? Not sure if we need this 🤔

File: cdk/lib/cdk-stack.ts (hunk 1 above)
Comment (agsaghhk) [reply cdk/lib/cdk-stack.ts:13]: Yeah currently this was just to create a baseline CDK setup. None of this is set in stone even after it is deployed to AWS.

File: cdk/lib/cdk-stack.ts (hunk 2 above)
Comment (agsaghhk) [reply cdk/lib/cdk-stack.ts:18]: Oh I didn't put the comment but I believe both of these should be default values. We can reduce the `receiveMessageWatiTime` to whatever we determine is best.

Comment (SaxyPandaBear) [reply cdk/lib/cdk-stack.ts:18]: If it's default, then whatever. Just figured I'd ask.

**Review by SaxyPandaBear — APPROVED** [review]

**Pull request merged**